### ISSUE TITLE
Is this package still in ongoing development?

### ISSUE THREAD

**Issue by pietz** [minimaxir/simpleaichat#92]
I really like the philosophy and design of this package but it appears its development process has basically stopped. I don't mean to pressure or to be rude but if @minimaxir could share his thoughts on this I would really appreciate it.

- Can we expect development by the maintainer?
- How can the community support?
- Are you open to PRs and reviewing them?
- What's your time commitment?
- Should someone just fork the repo and develop it from there?

Thanks

**Comment by minimaxir** [comment]
To be perfectly clear, this is absolutely pressure and being rude. My day-to-day job that actually pays me a salary wouldn't ever provide a bulleted list of the reasons I suck, let alone a project I develop in my spare time.

As mentioned in #91, I had a mental breakdown which is why I had to take a break and am just returning to work on it now. The reason I took a break is because there's nothing _broken_ with simpleaichat currently, nor are there any _urgent_ issues. I intentionally designed simpleaichat to have minimal dependencies and scope to avoid those issues and that has worked so far (OpenAI's demo day this week affected a bit but I'll add a few Issues for that).

Large open source AI organizations like Hugging Face and LangChain have created a false sense that open-source AI projects have to always be shipping 🚀🚀🚀. The difference is that both of those are maintained by those who do it as their full-time job and the companies are backed by significant venture capital. If I could devote 100% of my professional time to simpleaichat and other open-source projects, I would, but I do need a salary.

You also don't need to ask for permission for forking if you think it is necessary. That's the beauty of open source with permissive licensing. If a fork is better than what is the case now due to my inactivity, that's a good thing for everyone!

tl;dr as the MIT License that this repo follows says:

> THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND

**Comment by pors** [comment]
Hey Max, sorry to hear about your mental breakdown; I hope things are going better now!

You clearly created a library that hits a nerve with many developers; 3k stars is pretty impressive! 
The misunderstanding here is (I think) that people using this library run into issues and would like them solved and are (most of the time) happy to create a PR to resolve it. My impression is that you prefer to keep control of the code (please correct me if I'm wrong). If that is indeed the case, just say so in the README. That sets the expectations for this project straight and will offload any stress this might cause you.

Just my 2c <3

**Comment by minimaxir** [comment]
Everyone seems to think that there's such a thing as GitHub Issues-zero or PR-zero, which is an impossible standard. Every nontrivial open source project will have a queue, which necessitates a triage priority. Not all issues are equal.

I have accepted and merged PRs which have fixed issues / errors. That is not the case with every PR or open issue, e.g. the open ones regarding support to add Azure OpenAI support which is not as important of a feature and require significant QA work. There is a very big difference between important and *urgent*. I've done emergency fixes (and taken vacation time for it) for issues such as the Pydantic 2.0 migration, because that indeed caused breaking issues.

Again, unlike Hugging Face and LangChain repos, this project is very intentionally limited in scope to decrease the needed effort to maintain it, and I have refused to make additions which will complicate it.

I have responded to questions in GitHub issues with suggestions / ideas for a PR, and that's worked fine. With any open source project, there's always the possibility that an issue won't be followed-up on or a PR gets rejected: that's just the industry.

**Comment by pietz** [comment]
Sorry, to hear about your situatio, Max. I really tried to be as soft and nice in my initial message while still getting my question across. I even asked how the community can support. I still think these are fair things to ask especially if I see a patreon link and sponsor button.

Anyway, I hear what you said and I disagree with many of them. You built a great library but I think it doesn't align with a project I want to support at the moment.

All the best to you and your projects.

**Comment by minimaxir** [comment]
Hi @pietz , I apologize for my emotional response to your question.

I've clarified my stance in #95 , which is materially similar to what I posted here so will likely not change your opinion.
